tests/test_transforms/test_timestamp/test_event_transform.py

```diff
@@ -0,0 +1,164 @@
+    periods = 10
+    dataframes = [pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)}) for _ in range(3)]
+    for i in range(len(dataframes)):
+        df = dataframes[i]
+        df["segment"] = [f"segment_{i}"] * periods
+        df["target"] = np.arange(periods)
+        if i == 0:
+            df["holiday"] = np.array([0, 1, 0, 1, 1, 0, 0, 0, 1, 0])
+            df["holiday_binary_prev"] = np.array([1, 0, 1, 0, 0, 0, 1, 1, 0, 0])
+            df["holiday_binary_post"] = np.array([0, 0, 1, 0, 0, 1, 1, 0, 0, 1])
+            df["holiday_distance_prev"] = np.array([1, 0, 1, 0, 0, 0, 0.5, 1, 0, 0])
+            df["holiday_distance_post"] = np.array([0, 0, 1, 0, 0, 1, 0.5, 0, 0, 1])
+        elif i == 1:
+            df["holiday"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_binary_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_binary_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+        else:
+            df["holiday"] = np.array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+            df["holiday_binary_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_binary_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_prev"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            df["holiday_distance_post"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+    df = pd.concat(dataframes)
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D", known_future="all")
+    return tsds
+
+
+@pytest.fixture
+def ts_check_event_transform() -> TSDataset:
+    periods = 10
+    dataframes = [pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)}) for _ in range(3)]
+    for i in range(len(dataframes)):
+        df = dataframes[i]
+        df["segment"] = [f"segment_{i}"] * periods
+        df["target"] = np.arange(periods)
+        if i == 0:
+            df["holiday"] = np.array([0, 1, 0, 1, 1, 0, 0, 0, 1, 0])
+        elif i == 1:
+            df["holiday"] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+        else:
+            df["holiday"] = np.array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+    df = pd.concat(dataframes)
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D", known_future="all")
+    return tsds
+
+
+@pytest.fixture
+def ts_check_input_column_not_binary() -> TSDataset:
+    periods = 5
+    df = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)})
+    df["segment"] = ["segment_1"] * periods
+    df["target"] = np.arange(periods)
+    df["holiday"] = np.array([0, 1, 0, 0, 0.3])
+
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D")
+    return tsds
+
+
+def test_fit_transform(ts_check_event_transform: TSDataset, ts_check_event_transform_expected: TSDataset):
```

Comment:
Try to rework this test, it should be parametrized with
- mode
- n_pre
- n_post